src/dbnode/storage/series/types.go

```diff
@@ -283,3 +329,42 @@ func NewStats(scope tally.Scope) Stats {
 func (s Stats) IncCreatedEncoders() {
 	s.encoderCreated.Inc(1)
 }
+
+// WriteType is an enum for warm/cold write types.
+type WriteType int
+
+const (
+	// UndefinedWriteType is an undefined write type.
+	UndefinedWriteType WriteType = iota
+
+	// WarmWrite represents warm writes (within the buffer past/future window).
+	WarmWrite
+
+	// ColdWrite represents cold writes (outside the buffer past/future window).
+	ColdWrite
+)
+
+// WriteOptions define different options for a write.
+type WriteOptions struct {
```

Comment:
can we follow other Options pattern, where there is an interface, and use a package private struct to implement the WriteOptions interface? 

Consider making WriteOptions immutable as well, where SetXXX will produce a new option.

Reply:
Agreed. This seems like a simple enough change; I'll make these changes today and ping you when done.